iaso/models/stocks.py

```diff
@@ -0,0 +1,306 @@
+import typing
+import uuid
+
+from django.contrib.auth.models import AnonymousUser, User
+from django.db import models
+from django.db.models import QuerySet
+
+from iaso.models import Project
+from iaso.models.entity import UserNotAuthError
+from iaso.utils.models.soft_deletable import SoftDeletableModel
+
+
+class StockImpacts(models.TextChoices):
+    """StockImpacts is an Enum-like class representing the impact of a rule on the stock."""
+
+    ADD = "ADD", "Add"
+    SUBTRACT = "SUBTRACT", "Subtract"
+    RESET = "RESET", "Reset"
+
+
+class StockKeepingUnitQuerySet(QuerySet):
+    def filter_for_user(self, user: typing.Optional[typing.Union[User, AnonymousUser]]):
+        if not user or not user.is_authenticated:
+            raise UserNotAuthError("User not Authenticated")
+
+        profile = user.iaso_profile
+        return self.filter(account=profile.account)
+
+    def filter_for_user_and_app_id(self, user: typing.Optional[typing.Union[User, AnonymousUser]], app_id: str):
+        project = Project.objects.get_for_user_and_app_id(user, app_id)
+        return self.filter_for_project(project)
+
+    def filter_for_project(self, project: Project):
+        return self.filter(
+            account=project.account,
+            projects__in=[project],
+        )
+
+
+class StockKeepingUnit(SoftDeletableModel):
+    """
+    A distinct type of item for tracking in inventory. This is often referred as SKU.
+    """
+
+    name = models.CharField(max_length=255)
+    short_name = models.CharField(max_length=255)
+    account = models.ForeignKey("iaso.Account", on_delete=models.CASCADE, null=False, blank=False)
+    projects = models.ManyToManyField("Project", related_name="sku_projects", blank=True)
+    org_unit_types = models.ManyToManyField("OrgUnitType", related_name="sku_org_unit_types", blank=True)
+    forms = models.ManyToManyField("Form", related_name="sku_forms", blank=True)
+    children = models.ManyToManyField("self", blank=True, through="StockKeepingUnitChildren")
+    display_units = models.CharField(max_length=25, null=True, blank=True)
```

Comment:
Django considers [a bad practice](https://docs.djangoproject.com/en/5.2/ref/models/fields/#django.db.models.Field.null) to have both `null=True` and `blank=True` on a `CharField`

```suggestion
    display_units = models.CharField(max_length=25, blank=True)
```